### PR TITLE
Agent owns its directories

### PR DESCRIPTION
**Pull request by michalpristas**
## What does this PR do?

This changes owner of agent directories. Tested locally and on azure container instances, seems to work. 

## Why is it important?

Containers running on Azure Container Instances refuse to read config.

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `./changelog/fragments` using the [changelog tool](https://github.com/elastic/elastic-agent#changelog)
- [ ] I have added an integration test or an E2E test

### How to test

- build docker image
- push to observability-ci container repo
- create Azure (portal.azure.com login using SSO) Container Instance using published repo (you need to specify Other-Private docker registry with login and pass capture from docker-auth.elastic.co)
- see it runs

Closes: https://github.com/elastic/elastic-agent/issues/82

### CODE REVIEW

File: dev-tools/packaging/templates/docker/Dockerfile.elastic-agent.tmpl [hunk 1]
```diff
@@ -144,8 +143,10 @@ true
 RUN mkdir /app
 {{- end }}
 {{- else }}
-RUN groupadd --gid 1000 {{ .BeatName }}
-RUN useradd -M --uid 1000 --gid 1000 --groups 0 --home {{ $beatHome }} {{ .user }}
+RUN groupadd --gid 1000 {{ .BeatName }} && \
+    useradd -M --uid 1000 --gid 1000 --groups 0 --home {{ $beatHome }} {{ .user }} && \
+    chown -R {{ .user }}:{{ .user }} {{ $beatHome }} && \
+    true
```
Comment (pchila) [inline dev-tools/packaging/templates/docker/Dockerfile.elastic-agent.tmpl:149]: Why do you need a `&& true` at the end ? :thinking:

**Review by pchila — APPROVED** [review]
LGTM

File: dev-tools/packaging/templates/docker/Dockerfile.elastic-agent.tmpl (hunk 1 above)
Comment (michalpristas) [reply dev-tools/packaging/templates/docker/Dockerfile.elastic-agent.tmpl:149]: convention & readability, easier to add other commands later without messing with `&& \` at the end

**Comment by jlind23** [comment]
buildkite test it

**Comment by 111andre111** [comment]
@michalpristas 
I tested now with the changes und Docker Desktop as well as Azure and it was working.
Some household questions:
1. Is this one still needed?
https://github.com/elastic/elastic-agent/blob/bbede6e6cbe20932052e9a350d35567c77d13652/dev-tools/packaging/templates/docker/Dockerfile.elastic-agent.tmpl#L29
Given we later do nevertheless a chown I assume that particular step is not needed.
2. Is there the `--groups 0` still needed here:
https://github.com/elastic/elastic-agent/blob/bbede6e6cbe20932052e9a350d35567c77d13652/dev-tools/packaging/templates/docker/Dockerfile.elastic-agent.tmpl#L147
Given we don't have root group nevertheless inherited from that command like mentioned in the original issue, is there any need to add this user to supplementary group root?

**Comment by michalpristas** [comment]
first one is needed because beats have a check to see that only root can change the config, some security measure.

second one is needed because without it npm install and check for synthetics fails. i tried to work around it but i was not successful. there's an npm install command with || option to check `/root/something` and without this npm install fails and this check fails as well because `/root`

**Comment by 111andre111** [comment]
@michalpristas 
Thanks for these backgrounds. These male semse.
Now the `/root` this is probably happening because of `npm i -g`
https://stackoverflow.com/questions/18088372/how-to-npm-install-global-not-as-root
The location could be probably changed by `npm config set prefix` but I don't want to mess around too much. Thanks for the explanation.
Just to raise attention I still found this issue once agent is working:
https://github.com/elastic/elastic-agent/issues/82#issuecomment-1638233164

**Pull request merged**